### ISSUE TITLE
Portable installers are not renamed in non-symlinked installations

### ISSUE THREAD

**Issue by mdanish-kh** [microsoft/winget-cli#3437]
### Brief description of your issue

In a non-symlinked installation i.e. installation from a user shell in non-developer mode or when `ArchiveBinariesDependOnPath: true`, WinGet does not rename the portable file to `PortableCommandAlias` / `Commands` value from the manifest.

### Steps to reproduce

a. #3436 (`winget install FiloSottile.mkcert`)

b. https://github.com/microsoft/winget-pkgs/issues/111454 (`winget install Microsoft.Sysinternals.ProcessExplorer`)

c. `winget install Microsoft.Sysinternals.TCPView` on an x64 machine since the x64 executable is named `TCPView64.exe`

### Expected behavior

The portable file to be renamed allowing user to invoke the application through the specified command in the manifest.

### Actual behavior

Portable file isn't renamed and users have to type the full exe name in order to invoke the app.

### Environment

```shell
Windows Package Manager (Preview) v1.6.1573-preview
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1992
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.21.1573.0
```

**Comment by mdanish-kh** [comment]
[Policy] Portable

**Comment by mdanish-kh** [comment]
[Policy] Portable

**Comment by mdanish-kh** [comment]
An edge case may be `Microsoft.Sysinternals.ZoomIt` which is a ZIP package that contains portables for all 3 arches named as:

ZoomIt.zip/
├─ ZoomIt64.exe
├─ ZoomIt.exe
├─ ZoomIt64a.exe/

In x64 systems, you'd want to use ZoomIt64.exe and rename it to ZoomIt.exe in the Packages Directory, but this rename would cause conflict with the existing portable (supported for a different arch) in the directory. Maybe delete the existing one and force rename changes in this case?

**Comment by Trenly** [comment]
This has been partially addressed with #3439 , which covers when it is a single portable file, or when it the NestedInstallerPath points to a portable at the root of a Zip file. If the portable is inside of a folder in a Zip file, it will not be renamed. More work needs to be done here, but at least some of the cases have been covered

**Comment by o-l-a-v** [comment]
Is my PR hit by this issue, or am I doing something wrong? 🤔

* <https://github.com/microsoft/winget-pkgs/pull/172047>

Relevant part of manifest:

```yaml
  - Architecture: x64
    InstallerUrl: https://github.com/Azure/bicep/releases/download/v0.29.47/bicep-win-x64.exe
    InstallerSha256: 69305478E4CBBB9C9C5D74F146D2877A7211A3512F2F08AF0B172A035443F9E9
    InstallerType: portable
    NestedInstallerFiles:
      - RelativeFilePath: bicep-win-x64.exe
        PortableCommandAlias: bicep
```

Testing manifest locally works, but the command added to PATH is "bicep-win-x64", not "bicep".

```pwsh
PS > Get-Command bicep | fl

Get-Command: The term 'bicep' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

PS > Get-Command bicep-win-x64 | fl

Name            : bicep-win-x64.exe
CommandType     : Application
Definition      : C:\Users\olav.birkeland\AppData\Local\Microsoft\WinGet\Links\bicep-win-x64.exe
Extension       : .exe
Path            : C:\Users\olav.birkeland\AppData\Local\Microsoft\WinGet\Links\bicep-win-x64.exe
FileVersionInfo : File:             C:\Users\olav.birkeland\AppData\Local\Microsoft\WinGet\Links\bicep-win-x64.exe
                  InternalName:     bicep.dll
                  OriginalFilename: bicep.dll
                  FileVersion:      0.29.47.4906
                  FileDescription:  bicep
                  Product:          bicep
                  ProductVersion:   0.29.47+132ade51bc
                  Debug:            False
                  Patched:          False
                  PreRelease:       False
                  PrivateBuild:     False
                  SpecialBuild:     False
                  Language:         Language Neutral

PS >
```

WinGet version 1.8.1911.

**Comment by Trenly** [comment]
> Is my PR hit by this issue, or am I doing something wrong? 🤔
> 
> * [Microsoft.Bicep v0.29.47: Add arm64 and x64 portable installers winget-pkgs#172047](https://github.com/microsoft/winget-pkgs/pull/172047)
> 
> Relevant part of manifest:
> 
> ```yaml
>   - Architecture: x64
>     InstallerUrl: https://github.com/Azure/bicep/releases/download/v0.29.47/bicep-win-x64.exe
>     InstallerSha256: 69305478E4CBBB9C9C5D74F146D2877A7211A3512F2F08AF0B172A035443F9E9
>     InstallerType: portable
>     NestedInstallerFiles:
>       - RelativeFilePath: bicep-win-x64.exe
>         PortableCommandAlias: bicep
> ```
> 
> Testing manifest locally works, but the command added to PATH is "bicep-win-x64", not "bicep".
> 
> ```powershell
> PS > Get-Command bicep | fl
> 
> Get-Command: The term 'bicep' is not recognized as a name of a cmdlet, function, script file, or executable program.
> Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
> 
> PS > Get-Command bicep-win-x64 | fl
> 
> Name            : bicep-win-x64.exe
> CommandType     : Application
> Definition      : C:\Users\olav.birkeland\AppData\Local\Microsoft\WinGet\Links\bicep-win-x64.exe
> Extension       : .exe
> Path            : C:\Users\olav.birkeland\AppData\Local\Microsoft\WinGet\Links\bicep-win-x64.exe
> FileVersionInfo : File:             C:\Users\olav.birkeland\AppData\Local\Microsoft\WinGet\Links\bicep-win-x64.exe
>                   InternalName:     bicep.dll
>                   OriginalFilename: bicep.dll
>                   FileVersion:      0.29.47.4906
>                   FileDescription:  bicep
>                   Product:          bicep
>                   ProductVersion:   0.29.47+132ade51bc
>                   Debug:            False
>                   Patched:          False
>                   PreRelease:       False
>                   PrivateBuild:     False
>                   SpecialBuild:     False
>                   Language:         Language Neutral
> 
> PS >
> ```
> 
> WinGet version 1.8.1911.

No. It looks like you’re trying to use `NestedInstallerFiles` which is only valid on `InstallerType: zip`. You should remove that section and instead use
```
Commands:
- bicep
- ```

**Comment by o-l-a-v** [comment]
> It looks like you’re trying to use `NestedInstallerFiles` which is only valid on `InstallerType: zip`. You should remove that section and instead use
> 
> ```
> Commands:
> - bicep
> ```

Aha! Sweet, thanks @Trenly, that worked. 🙂

**Comment by mdanish-kh** [comment]
This issue is more exaggerated now when the introduction of `ArchiveBinariesDependOnPath: true` (https://github.com/microsoft/winget-cli/pull/4816). All zip-binary packages will face this issue since those installations are always non-symlink

See also: https://github.com/microsoft/winget-pkgs/issues/233868

**Comment by mdanish-kh** [comment]
This issue is more exaggerated now with the introduction of `ArchiveBinariesDependOnPath: true` (https://github.com/microsoft/winget-cli/pull/4816). All zip-binary packages will face this issue since those installations are always non-symlink

See also: https://github.com/microsoft/winget-pkgs/issues/233868

**Comment by ririshi** [comment]
Is `ArchiveBinariesDependOnPath` actually documented anywhere other than in GH issues? I've spent a good amount of time trying to figure out why winget won't create a symlink for `scrcpy`. I thought it was a bug or a problem with my setup at first. It would be really nice if the installer output made this behaviour explicit. Right now it only says "Command line alias added: scrcpy", even though I expected a symlinked installation because of my admin/dev-mode shell.

(sorry if this is too off-topic for this issue, I just came here while on my journey to figure out what was happening to my scrpy installation)
